app/screens/send-bitcoin-screen/payment-details/index.types.ts

```diff
@@ -0,0 +1,116 @@
+import {
+  GraphQlApplicationError,
+  PaymentSendResult,
+  useIntraLedgerPaymentSendMutation,
+  useIntraLedgerUsdPaymentSendMutation,
+  useLnInvoiceFeeProbeMutation,
+  useLnInvoicePaymentSendMutation,
+  useLnNoAmountInvoiceFeeProbeMutation,
+  useLnNoAmountInvoicePaymentSendMutation,
+  useLnNoAmountUsdInvoiceFeeProbeMutation,
+  useLnNoAmountUsdInvoicePaymentSendMutation,
+  useLnUsdInvoiceFeeProbeMutation,
+  useOnChainPaymentSendMutation,
+  useOnChainTxFeeLazyQuery,
+  WalletCurrency,
+} from "@app/graphql/generated"
+import { BtcPaymentAmount, PaymentAmount } from "@app/types/amounts"
+import { WalletDescriptor } from "@app/types/wallets"
+import { LnUrlPayServiceResponse } from "lnurl-pay/dist/types/types"
+
+export type ConvertPaymentAmount = <T extends WalletCurrency>(
+  paymentAmount: PaymentAmount<WalletCurrency>,
+  toCurrency: T,
+) => PaymentAmount<T>
+
+export type BaseCreatePaymentDetailsParams<T extends WalletCurrency> = {
+  convertPaymentAmount: ConvertPaymentAmount
+  sendingWalletDescriptor?: WalletDescriptor<T>
+  destinationSpecifiedMemo?: string
+  senderSpecifiedMemo?: string
+}
+
+export type SetSendingWalletDescriptor<T extends WalletCurrency> = (
+  sendingWalletDescriptor: WalletDescriptor<T>,
+) => PaymentDetail<T>
+
+export type SetUnitOfAccount<T extends WalletCurrency> = (
+  unitOfAccount: WalletCurrency,
+) => PaymentDetail<T>
+
+type GetFeeParams = {
+  lnInvoiceFeeProbe: ReturnType<typeof useLnInvoiceFeeProbeMutation>["0"]
+  lnNoAmountInvoiceFeeProbe: ReturnType<typeof useLnNoAmountInvoiceFeeProbeMutation>["0"]
+  lnNoAmountUsdInvoiceFeeProbe: ReturnType<
+    typeof useLnNoAmountUsdInvoiceFeeProbeMutation
+  >["0"]
+  lnUsdInvoiceFeeProbe: ReturnType<typeof useLnUsdInvoiceFeeProbeMutation>["0"]
+  onChainTxFee: ReturnType<typeof useOnChainTxFeeLazyQuery>["0"]
+}
+
+export type GetFee<T extends WalletCurrency> = (getFeeFns: GetFeeParams) => Promise<{
+  amount?: PaymentAmount<T> | null | undefined
+  errors?: readonly GraphQlApplicationError[]
+}>
+
+type SendPaymentParams = {
+  lnInvoicePaymentSend: ReturnType<typeof useLnInvoicePaymentSendMutation>["0"]
```

Comment:
can't you use this type directly? `LnInvoicePaymentSendMutationResult` or `LnInvoicePaymentSendMutationHookResult` (same for all the other types here)

Reply:
You could, although we just need the function, so it is not necessary to depend on the entire return result. It also makes mocking a little simpler for the test by just having to mock the function.